packages/neos-ui/src/Sagas/Changes/index.js

```diff
@@ -9,6 +9,12 @@ const {publishableNodesInDocumentSelector, baseWorkspaceSelector} = selectors.CR
 function * persistChanges(changes) {
     const {change} = backend.get().endpoints;
 
+    const confirmExit = () => {
+        return 'You have attempted to leave this page. Your changes are currently persisted, if you leave now your changes will may not be saved.  Are you sure you want to exit this page?';
```

Comment:
For some browsers you also need to assign return value to the event, e.g. `e.returnValue = dialogText;`
See https://developer.mozilla.org/en-US/docs/Web/API/WindowEventHandlers/onbeforeunload

Reply:
Also that page says that it's better to use `window.addEventListener` to bind the handler.